lib/stringref.h

```diff
@@ -51,11 +51,14 @@ struct StringRef {
         len = 0;
     }
     StringRef(const StringRef &a) : p(a.p), len(a.len) {}
+    // avoid clang-tidy complaining
+    // NOLINTBEGIN(bugprone-unhandled-self-assignment)
```

Comment:
```
    if(this == &a)
      return *this;
```
instead?

Reply:
This one is safe to be performed, I'm not sure if the if is worth having in such cases. If the if is there, one would have to wonder why it is needed.